features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdIpManagerDaoImpl.java

```diff
@@ -79,58 +82,63 @@ public synchronized void dataSourceSync() throws SQLException {
     	m_knownips.clear();
     	for (OnmsIpInterface one: list) {
     		if (one != null) {
-    			m_knownips.put(InetAddressUtils.str(one.getIpAddress()), (long) one.getId());
+    			m_knownips.put(one.getIpAddress(), one.getId());
```

Comment:
This is not correct: we want the ID of the *OnmsNode* as the value in the map, not the ID of the OnmsIpInterface. Please replace this method with the original copy from HibernateTrapdIpMgr:
```
m_knownips = m_ipInterfaceDao.getInterfacesForNodes();
```